pop_tools/modify_pop_for_xgcm.py

```diff
@@ -0,0 +1,97 @@
+"""
+Module for modifying pop model output to be compatible with xgcm
+
+# Ref: https://gist.github.com/rabernat/933bc785c99828352f343e48d0da6e22
+"""
+
+import numpy as np
+import xarray as xr
+
+
+def _add_pop_dims_to_dataset(ds):
+    ds_new = ds.copy()
+    ds_new['nlon_u'] = xr.Variable(
+        ('nlon_u'), np.arange(len(ds.nlon)) + 1, {'axis': 'X', 'c_grid_axis_shift': 0.5}
+    )
+    ds_new['nlat_u'] = xr.Variable(
+        ('nlat_u'), np.arange(len(ds.nlat)) + 1, {'axis': 'Y', 'c_grid_axis_shift': 0.5}
+    )
+    ds_new['nlon_t'] = xr.Variable(('nlon_t'), np.arange(len(ds.nlon)) + 0.5, {'axis': 'X'})
+    ds_new['nlat_t'] = xr.Variable(('nlat_t'), np.arange(len(ds.nlat)) + 0.5, {'axis': 'Y'})
+
+    # add metadata to z grid
+    ds_new['z_t'].attrs.update({'axis': 'Z'})
+    ds_new['z_w'].attrs.update({'axis': 'Z', 'c_grid_axis_shift': -0.5})
+    ds_new['z_w_top'].attrs.update({'axis': 'Z', 'c_grid_axis_shift': -0.5})
+    ds_new['z_w_bot'].attrs.update({'axis': 'Z', 'c_grid_axis_shift': 0.5})
+
+    return ds_new
+
+
+def _dims_from_grid_loc(grid_loc):
+    grid_loc = str(grid_loc)
+    ndim = int(grid_loc[0])
+    x_loc_key = int(grid_loc[1])
+    y_loc_key = int(grid_loc[2])
+    z_loc_key = int(grid_loc[3])
+
+    x_loc = {1: 'nlon_t', 2: 'nlon_u'}[x_loc_key]
```

Comment:
You also need

```
x_loc = {1: "nlon_t", 2: "nlon_u", 3: "nlon_u"}[x_loc_key]
y_loc = {1: "nlat_t", 2: "nlat_u", 3: "nlat_t"}[y_loc_key]
```

to make `DZU` and `DZT` work

Reply:
Sounds good. Thank you, @dcherian 

Can you point me to a small test dataset that I can use for testing purposes?

Reply:
Ask Anna on that email thread. I just left work